### PR TITLE
Spark 3.1.1 support

### PR DESCRIPTION
**Pull request by rozza**
SPARK-293

https://spruce.mongodb.com/version/607ecc385623436443a9f4c6

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Mongo Spark Connector Changelog
 
+## 3.1.0
```
Comment (jaceklaskowski) [inline CHANGELOG.md:3]: nit: 3.1.1? Why not to match Spark version?

**Review by jaceklaskowski — COMMENTED** [review]
I must admit I'm surprised how few changes were proposed to support Spark 3.1.1. Nothing bad but expected `Table` and such. Anyway...

File: CHANGELOG.md (hunk 1 above)
Comment (rozza) [reply CHANGELOG.md:3]: Because this is the first release of the 3.1 series and there have been no patches to the connector.

**Comment by rozza** [comment]
@jaceklaskowski my tests have shown the 3.0.1 spark connector works against 3.1.1 Spark. Which is how semantic versioning should work but unfortunately hasn't always been the case with previous minor spark releases.

**Pull request closed without merging**